activerecord/lib/active_record/railtie.rb

```diff
@@ -157,9 +157,16 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
-    initializer "active_record.set_executor_hooks" do
+    initializer "active_record.set_executor_hooks" do |app|
+      conn_hook = ActiveRecord::ConnectionAdapters::ConnectionManagement.new
+
       ActiveSupport.on_load(:active_record) do
         ActiveRecord::QueryCache.install_executor_hooks
+        app.executor.register_hook(conn_hook)
+      end
+
+      ActiveSupport.on_load(:action_dispatch_integration_test) do
```

Comment:
I'm not sure it's we want to disable it globally when ActionDispatch::IntegrationTest loaded.
For example we want to phantomjs end to end tests (clean_connections desired) as well as integration test (no clean connections) in the same run.